vulture/config.py

```diff
@@ -0,0 +1,212 @@
+DEFAULTS = {
+    "min_confidence": 0,
+    "paths": [],
+    "exclude": [],
+    "ignore_decorators": [],
+    "ignore_names": [],
+    "make_whitelist": False,
+    "sort_by_size": False,
+    "verbose": False,
+}
+
+#: sentinel value to distinguish between "False" and "no default given"
+MISSING = object()
+
+
+def _check_input_config(data):
+    """
+    Checks the types of the values in *data* against the expected types of
+    config-values. If a value is of the wrong type it will raise a SystemExit.
+    """
+    for key, value in data.items():
+        if key not in DEFAULTS:
+            print(f"Unknown configuration key: {key}", file=sys.stderr)
+            sys.exit("Invalid config")
+        if value is not MISSING and not type(value) is type(  # noqa: E721
+            DEFAULTS[key]
+        ):
+            print(
+                f"Data type for {key} must be {DEFAULTS[key]}", file=sys.stderr
+            )
+            sys.exit("Invalid config")
+
+
+def from_dict(data):
+    """
+    Create a new config dictionary from an existing one, assign possible
+    defaults and warn about unprocessed options.
+    """
+    _check_input_config(data)
+
+    unknown_keys = set(data) - set(DEFAULTS)
+    if unknown_keys:
+        print(
+            f"Unknown configuration keys: {sorted(unknown_keys)}",
+            file=sys.stderr,
+        )
+        sys.exit("invalid Config")
+
+    output = {}
+    for key, default in DEFAULTS.items():
+        output[key] = data.get(key, default)
+    return output
+
+
+def _parse_toml(infile):
+    """
+    Parse a TOML file for config values.
+
+    It will search for a section named ``[tool.vulture]`` which contains the
+    same keys as the CLI arguments seen with ``--help``. All leading dashes are
+    removed and other dashes are replaced by underscores (so ``--sort-by-size``
+    becomes ``sort_by_size``).
+
+    Arguments containing multiple values are standard TOML lists.
+
+    Example::
+
+        [tool.vulture]
+        exclude = ["file*.py", "dir/"]
+        ignore_decorators = ["deco1", "deco2"]
+        ignore_names = ["name1", "name2"]
+        make_whitelist = true
+        min_confidence = 10
+        sort_by_size = true
+        verbose = true
+        paths = ["path1", "path2"]
+    """
+    data = toml.load(infile)
+    vulture_settings = data.get("tool", {}).get("vulture", {})
+    return from_dict(vulture_settings)
+
+
+def _parse_args(args=None):
+    """
+    Parse CLI arguments
+
+    :param args: A list of strings representing the CLI arguments. If left to
+        the default, this will default to ``sys.argv``.
+    """
+
+    def csv(exclude):
+        return exclude.split(",")
+
+    usage = "%(prog)s [options] PATH [PATH ...]"
+    version = f"vulture {__version__}"
+    glob_help = "Patterns may contain glob wildcards (*, ?, [abc], [!abc])."
+    parser = argparse.ArgumentParser(prog="vulture", usage=usage)
+    parser.add_argument(
+        "paths",
+        nargs="+",
```

Comment:
The "+" requires users to specify paths on the command line. They can't do it in the TOML file. I think we should remove this requirement and update the `usage` string above on the way. Also, we should check that after reading all config values, there is at least one path and exit with a useful error message otherwise (if that isn't done already). Finally, this argument should also get `default=MISSING`.